pandas/core/arrays/masked.py

```diff
@@ -305,3 +329,24 @@ def value_counts(self, dropna: bool = True) -> "Series":
         counts = IntegerArray(counts, mask)
 
         return Series(counts, index=index)
+
+    def _reduce(self, name: str, skipna: bool = True, **kwargs):
+        data = self._data
+        mask = self._mask
+
+        if name in {"sum", "prod", "min", "max"}:
+            op = getattr(masked_reductions, name)
+            return op(data, mask, skipna=skipna, **kwargs)
+
+        # coerce to a nan-aware float if needed
+        # (we explicitly use NaN within reductions)
+        if self._hasna:
+            data = self.to_numpy("float64", na_value=np.nan)
+
+        op = getattr(nanops, "nan" + name)
+        result = op(data, axis=0, skipna=skipna, mask=mask, **kwargs)
+
+        if np.isnan(result):
+            return libmissing.NA
```

Comment:
as a follow up, should this be self.dtype.na_value instead of hardcoded in the base class.

Reply:
I personally like having it explicit as well (all subclasses use `NA` as missing value indicator, this method is not shared with ones where `dtype.na_value` is something different), but can go either way

Reply:
hmm,

we have 
```
    @property
    def _na_value(self):
        return self.dtype.na_value
```

so this is not consistent.

Reply:
We could also remove that property, from a quick search, that is only used once in the arrays code

Reply:
I agree, if not generally on an array class, I think self.dtype.na_value is generally clearer.

This is not the only case where libmissing.NA is used, so we should use one ore the other. I accept your argument about being explicit but also prefer the the case of allowing the class to be extensible. YAGNI favours your proposal.